server.js

```diff
@@ -51,37 +61,72 @@ let messageList = [];
 const addToList = message => messageList.push(message);
 subscribers.push(addToList);
 
-// https://expressjs.com/en/starter/basic-routing.html
-app.get("/", (request, response) => {
-  response.sendFile(__dirname + "/views/index.html");
-});
+const routeGetRootWithUserId = config => (request, response) => {
+  let template = PageWrapper({
+    title: config.pageTitle,
+    children: ChatRoom({
+      userName: request.params.id ? request.params.id : config.defaultUserName
+    })
+  });
+  response.send(template);
+}
 
-app.get("/messages", (request, response) => {
-  response.type("text/plain");
+const routeGetRoot = config => (request, response) => {
+  let template = PageWrapper({
+    title: config.pageTitle,
+    children: SignIn()
+  });
+  response.send(template);
+}
+
+const routePostRoot = config => (request, response) => {
+  response.redirect(`/${request.body.user}`);
```

Comment:
When you post, we make a redirect to :id with the current hidden  "user" field that's provided in the chat.